x/programs/rust/examples/token/src/lib.rs

```diff
@@ -25,27 +38,38 @@ pub fn mint_to(state: State, recipient: Address, amount: i64) -> bool {
         .is_ok()
 }
 
-/// Transfers amount coins from the sender to the recipient. Returns whether successful.
+/// @notice Transfers coins from the sender to the recipient
+/// @param state The current program state
+/// @param sender The address sending tokens
+/// @param recipient The address receiving tokens
+/// @param amount The amount of tokens to transfer
+/// @return Whether the transfer succeeded
 #[public]
 pub fn transfer(state: State, sender: Address, recipient: Address, amount: i64) -> bool {
     // require sender != recipient
-    if sender == recipient {
-        return false;
-    }
+    assert_ne!(sender, recipient);
+
     // ensure the sender has adequate balance
     let sender_balance: i64 = state.get_map_value("balances", &sender).unwrap_or(0);
-    if amount < 0 || sender_balance < amount {
-        return false;
-    }
+    assert!(amount >= 0 && sender_balance >= amount);
+
     let recipient_balance: i64 = state.get_map_value("balances", &recipient).unwrap_or(0);
+
+    let sender_delta = sender_balance.checked_sub(amount);
+    let recipient_delta = recipient_balance.checked_add(amount);
+    assert!(sender_delta.is_some() && recipient_delta.is_some());
+
     state
-        .store_map_value("balances", &sender, &(sender_balance - amount))
-        .store_map_value("balances", &recipient, &(recipient_balance + amount))
+        .store_map_value("balances", &sender, &sender_delta.unwrap())
+        .store_map_value("balances", &recipient, &recipient_delta.unwrap())
```

Comment:
Using `Option::unwrap` is generally considered a bad practice.

Reply:
It's true but we assert that they are both `Some(T)` above. I don't see an elegant way to get to the inner value otherwise.

Reply:
Ok nvm found a way